app/scripts/modules/core/src/managed/overview/Resource.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+
+import { Icon, useApplicationContextSafe } from 'core/presentation';
+import { IconTooltip } from 'core/presentation/IconTooltip';
+
+import { MdResourceActuationState, useFetchResourceStatusQuery } from '../graphql/graphql-sdk';
+import spinner from './loadingIndicator.svg';
+import { ResourceTitle } from '../resources/ResourceTitle';
+import { IResourceLinkProps, resourceManager } from '../resources/resourceRegistry';
+import { QueryResource } from './types';
+import { TOOLTIP_DELAY } from '../utils/defaults';
+
+import './Resource.less';
+
+const statusUtils: {
+  [key in Exclude<MdResourceActuationState['status'], 'UP_TO_DATE'>]: {
+    color?: string;
+    icon: string;
+    defaultReason: string;
+  };
+} = {
+  ERROR: { color: 'var(--color-status-error)', icon: 'fas fa-times', defaultReason: 'Failed to update resource' },
+  NOT_MANAGED: { color: 'var(--color-status-warning)', icon: 'fas fa-pause', defaultReason: 'Resource is not manged' },
+  WAITING: { icon: 'far fa-hourglass', defaultReason: 'Resource is currently locked and can not be updated' },
+  PROCESSING: { icon: 'far fa-hourglass', defaultReason: 'Resource is being updated' },
+};
+
+const Status = ({
+  appName,
+  environmentName,
+  resourceId,
+}: {
+  appName: string;
+  environmentName: string;
+  resourceId: string;
+}) => {
+  const { data: resourceStatuses, error, loading } = useFetchResourceStatusQuery({ variables: { appName } });
+  const state = resourceStatuses?.application?.environments
+    .find((env) => env.name === environmentName)
+    ?.state.resources?.find((resource) => resource.id === resourceId)?.state;
+
+  if (error || (!loading && !state)) {
+    return (
```

Comment:
This is a bit different because `state` isn't the response but the state of the resource from the response. 
Yeah, we need to add a logging mechanism for that (we don't have radar on deck).